plugins/modules/dnf_config_manager.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Andrew Hyatt <andy@hyatt.xyz>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: dnf_config_manager
+short_description: Enable or disable dnf repositories using config-manager
+version_added: 8.2.0
+description:
+  - This module enables or disables repositories using the C(dnf config-manager) sub-command.
+author: Andrew Hyatt (@ahyattdev) <andy@hyatt.xyz>
+requirements:
+  - dnf
+  - dnf-plugins-core
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - Repository id, for example V(crb).
+    default: []
+    required: false
+    type: list
+    elements: str
+  state:
+    description:
+      - Whether the repositories should be V(enabled) or V(disabled).
+    default: enabled
+    required: false
+    type: str
+    choices: [enabled, disabled]
+'''
+
+EXAMPLES = r'''
+- name: Ensure the crb repository is enabled
+  community.general.dnf_config_manager:
+    name: crb
+    state: enabled
+
+- name: Ensure the appstream and zfs repositories are disabled
+  community.general.dnf_config_manager:
+    name:
+      - appstream
+      - zfs
+    state: present
+'''
+
+RETURN = r'''
+repo_states_pre:
+  description: Repo IDs before action taken.
+  returned: success
+  type: dict
+  contains:
+    repo_id:
+      description: Repository ID
+      type: str
+  sample: |-
```

Comment:
The type is dict, not string, so you should pass a dict.
```suggestion
  sample:
```